sound/hda/ext/hdac_ext_stream.c

```diff
@@ -106,20 +106,14 @@ void snd_hdac_stream_free_all(struct hdac_bus *bus)
 }
 EXPORT_SYMBOL_GPL(snd_hdac_stream_free_all);
 
-/**
- * snd_hdac_ext_stream_decouple - decouple the hdac stream
- * @bus: HD-audio core bus
- * @stream: HD-audio ext core stream object to initialize
- * @decouple: flag to decouple
- */
-void snd_hdac_ext_stream_decouple(struct hdac_bus *bus,
-				struct hdac_ext_stream *stream, bool decouple)
```

Comment:
do I understand it correctly, that this function uses a double-negation logic? It's called `decouple` (not-couple) and has a `decouple` (not-couple) parameter, so, calling `decouple(false)` means not-not-couple, i.e. do the coupling?.. Wouldn't it be better to split it into 2 functions: `couple()` and `decouple()` with no flags? There's practically 0 common code anyway. Then verifying locking should also become easier...